### ISSUE TITLE
What is the project roadmap

### ISSUE THREAD

**Issue by elmatou** [geospoc/v-mapbox#364]
Hi everyone, (@vinayakkulkarni )
I thank you for forking the original project and try to maintain it.

While trying to use it, I found that a lot of features from mapbox-gl are missing, reactivity is quite lacking. and some core concept seems to me a bit strange.

I'm willing to work on the lib, but I need to know if the following roadmap is acceptable to you, or if I'll have to fork the lib another time (that would be sad)
- Make every options in mapbox-gl a prop in v-mapbox (some are missing, I did some PR about that), and remove unnecessary default (mapbox-gl handle default values at is level).
- Make every props reactive (lots of watchers are missing).
- Simplify API ( eg: MglGeojsonLayer wait for layerId prop which is not consistent with mapbox layer API, where id is a property of layer object)
- refactor events to be closer to mapbox-gl genuine trigger and payload
- Support mapbox-gl from V1 until V2 (pre v1 support is a possibility) & maplibre

Of course if this fork is a company project, I will understand.
But there are lots of very good intents here, that need to be pushed a bit farther to be really useful to the community.

**Comment by Maker-Mark** [comment]
@elmatou I'm here too to work on this. I think the original project was a great start. I just found this repo, and I'm glad to see people maintaining it. I have had to use the mapbox api more than the vue-mapbox api in the past few weeks.

For me, the first thing would be to add all of the mapbox-gl props. There are so many important ones missing (geolocate, dragEnd, and much more). 

More watchers would be great, and hooks into the style.
Is this a fork for a company?

**Comment by elmatou** [comment]
Hi @Maker-Mark,

I haven't got a response from geospoc/v-mapbox maintainer yet.

If I get no response I'm willing to fork the repo :
- It would be a community driven project (I currently need to switch from leaflet for an aerial navigation soft).
- I intent to stick to mapbox/maplibre API, no fancy features.

**Comment by vinayakkulkarni** [comment]
Hey @elmatou & @Maker-Mark,

I plan to maintain this project (along with some of my co–workers here at @geospoc) and upgrade it to Vue 3 (with TS) as well as keep a Vue 2.x version (with TS). Currently been really packed with personal stuff (COVID related) and work (deliveries)

I do welcome community inputs & have merged few PRs already which I believe were add–ons to the project.

I'll be creating a project boards for `v2.x` & `v3.x` of this project where `v-mapbox v2.x` is for supporting `Vue 2.x + TS` & `v-mapbox v3.x` will be based on `Vue 3.x + TS`

**Comment by Maker-Mark** [comment]
@vinayakkulkarni Nice, I'd love to help to maintain this repo. I'll keep an eye out for the projects. Please @ me to get me involved on a ticket. Here to help!

**Comment by elmatou** [comment]
@vinayakkulkarni & @Maker-Mark 

I'm gonna PR some stuff, as soon as we validate a roadmap,
In the mean time, I'm gonna work on missing props or feature from the official API.

Do you really think TS is a game changer for a wrapper ?

**Comment by Maker-Mark** [comment]
@elmatou No, I dont think TS is a game changer for the wrapper. I'd prioritize documentation and implementation of the remaining methods on the wrapper. We don't have some core functionality like `@geolocate` and `@ondragend` wrapped and documented.
